### PR TITLE
address build warnings after async analyzer change and fix for Tenant in cache

### PR DESCRIPTION
**Pull request by jennyf19**

### CODE REVIEW

File: core/src/Cache/MsalAccountCacheItem.cs [hunk 1]
```diff
@@ -58,7 +58,7 @@ internal MsalAccountCacheItem(Authority authority, MsalTokenResponse response) :
             Environment = environment;
             PreferredUsername = preferredUsername;
             Name = name;
-            TenantId = TenantId;
+            TenantId = tenantId;
```
Comment (henrik-me) [inline core/src/Cache/MsalAccountCacheItem.cs:61]: >tenantI [](start = 23, length = 7)

:)

File: msal/samples/desktop/SampleApp/MainForm.cs [hunk 2]
```diff
@@ -58,19 +53,21 @@ public MainForm()
             }
         }
 
-        private async void TabControl1_SelectedIndexChanged(object sender, EventArgs e)
+        // Return void because this is an event handler
+        private async void TabControl1_SelectedIndexChangedAsync(object sender, EventArgs e)
```
Comment (henrik-me) [inline msal/samples/desktop/SampleApp/MainForm.cs]: >private async void TabControl1_SelectedIndexChangedAsync(object sender, EventArgs e) [](start = 8, length = 84)

these are event ui handlers and should probably just have been added to the ignore list. It's like the Main method in a console app. (doc for async will state something similar using better words)

Comment (henrik-me) [reply msal/samples/desktop/SampleApp/MainForm.cs]: (same issue in a few places, but most of the places are spot on, thanks a ton for doing this!)

---
In reply to: [202498640](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1132#discussion_r202498640) [](ancestors = 202498640)

File: msal/src/Microsoft.Identity.Client/UIParent.cs [hunk 3]
```diff
@@ -50,6 +50,9 @@ public UIParent()
         }
 
 #if iOS
+        /// <summary>
+        /// Constructor for iOS for implementing embedded webview
+        /// </summary>
```
Comment (henrik-me) [inline msal/src/Microsoft.Identity.Client/UIParent.cs:55]: not sure if you should add a remark (a real thing as part of the xml doc)/ saying this method is likely to be removed before final release

File: msal/samples/desktop/SampleApp/MainForm.cs (hunk 2 above)
Comment (henrik-me) [reply msal/samples/desktop/SampleApp/MainForm.cs]: btw.... now you have done it I recommend leaving it as is.

---
In reply to: [202498676](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1132#discussion_r202498676) [](ancestors = 202498676,202498640)

**Review by henrik-me — APPROVED** [review]
:shipit:

File: msal/src/Microsoft.Identity.Client/UIParent.cs (hunk 3 above)
Comment (jennyf19) [reply msal/src/Microsoft.Identity.Client/UIParent.cs:55]: done.

File: msal/samples/desktop/SampleApp/MainForm.cs (hunk 2 above)
Comment (jennyf19) [reply msal/samples/desktop/SampleApp/MainForm.cs]: i fixed all the event handlers, but you'll have to show me how to get it to ignore those methods so we can remove `async` in them too. they don't show up in MSBuild, just in VS as warnings.

**Pull request merged**